src/containers/Tenant/Diagnostics/Configs/Configs.tsx

```diff
@@ -0,0 +1,149 @@
+import type {Column} from '@gravity-ui/react-data-table';
+import {Icon, Popover, Switch} from '@gravity-ui/uikit';
+import {StringParam, useQueryParam} from 'use-query-params';
+
+import {ResizeableDataTable} from '../../../../components/ResizeableDataTable/ResizeableDataTable';
+import {Search} from '../../../../components/Search';
+import {TableWithControlsLayout} from '../../../../components/TableWithControlsLayout/TableWithControlsLayout';
+import {setFeatureFlagsFilter, tenantApi} from '../../../../store/reducers/tenant/tenant';
+import type {TConfigFeatureFlag} from '../../../../types/api/featureFlags';
+import {cn} from '../../../../utils/cn';
+import {DEFAULT_TABLE_SETTINGS} from '../../../../utils/constants';
+import {useTypedDispatch, useTypedSelector} from '../../../../utils/hooks';
+
+import i18n from './i18n';
+
+import './Configs.scss';
+
+const FEATURE_FLAGS_COLUMNS_WIDTH_LS_KEY = 'featureFlagsColumnsWidth';
+
+const b = cn('ydb-diagnostics-configs');
+
+const columns: Column<TConfigFeatureFlag>[] = [
+    {
+        name: ' ',
+        render: ({row}) => (
+            <React.Fragment>
+                {row.Current && (
+                    <Popover
+                        content={i18n('flag-touched')}
+                        className={b('icon-touched')}
+                        placement="left"
+                    >
+                        <Icon data={PersonPencil} />
+                    </Popover>
+                )}
+            </React.Fragment>
+        ),
+        width: 36,
+        sortable: false,
+        resizeable: false,
+    },
+    {
+        name: i18n('td-feature-flag'),
+        render: ({row}) => (row.Current ? <b>{row.Name}</b> : row.Name),
+        width: 400,
+        sortable: true,
+        sortAccessor: ({Current, Name}) => {
+            return Number(!Current) + Name.toLowerCase();
+        },
+    },
+    {
+        name: i18n('td-default'),
+        render: ({row}) => {
+            switch (row.Default) {
+                case true:
+                    return i18n('enabled');
+                case false:
+                    return i18n('disabled');
+                default:
+                    return '-';
+            }
+        },
+        width: 100,
+        sortable: false,
+        resizeable: false,
+    },
+    {
+        name: i18n('td-current'),
+        render: ({row}) => <Switch disabled checked={(row.Current ?? row.Default) || false} />,
+        width: 100,
+        sortable: false,
+        resizeable: false,
+    },
+];
+
+interface Props {
+    path: string;
+    database: string;
+}
+
+export const Configs = ({database}: Props) => {
+    const dispatch = useTypedDispatch();
+    const [search, setSearch] = useQueryParam('search', StringParam);
+    const {currentData = [], isFetching, isError} = tenantApi.useGetClusterConfigQuery({database});
+    const featureFlagsFilter = useTypedSelector(
+        (state) => state.tenant.featureFlagsFilter,
+    )?.toLocaleLowerCase();
```

Comment:
Do we really need this state in the store?

Reply:
yes. in some other tabs search filter is saved between tab navigation

Reply:
But current solution is not good, we have two sources of truth. On other tabs behavior is different. For example, on Top queries tab the search is saved only in the store, and on Nodes and Storage tabs the search is saved only in query params.

Reply:
I suggested to have same UX in such cases - sync btw store and url, others agreed.
Why do you think it's not good?

Reply:
discussed in pm, fixed